File: Python/1022 - TDA racional.py
n = int(input())

for x in range(n):
    info = input().split(' ')

    n1 = int(info[0])
    d1 = int(info[2])
    n2 = int(info[4])
    d2 = int(info[6])

    if info[3] == "+":
        a = n1*d2+n2*d1
        b = d1*d2

        r1 = str(a) + "/" + str(b)

        i = 2

        while True:
            if a % i == 0 and b % i == 0:
                a = a/i
                b = b/i
            elif i > a or i > b:
                break
            else:
                i += 1

        r1 += " = " + str(int(a)) + "/" + str(int(b))
        print(r1)

    elif info[3] == "-":
        a = n1*d2-n2*d1
        b = d1*d2

        r1 = str(a) + "/" + str(b)

        i = 2

        while True:
            if a % i == 0 and b % i == 0:
                a = a/i
                b = b/i
            elif i > a or i > b:
                break
            else:
                i += 1

        r1 += " = " + str(int(a)) + "/" + str(int(b))
        print(r1)

    elif info[3] == "*":
        a = n1*n2
        b = d1*d2

        r1 = str(a) + "/" + str(b)

        i = 2

        while True:
            if a % i == 0 and b % i == 0:
                a = a/i
                b = b/i
            elif i > a or i > b:
                break
            else:
                i += 1

        r1 += " = " + str(int(a)) + "/" + str(int(b))
        print(r1)

    elif info[3] == "/":
        a = n1*d2
        b = n2*d1

        r1 = str(a) + "/" + str(b)

        i = 2

        while True:
            if a % i == 0 and b % i == 0:
                a = a/i
                b = b/i
            elif i > a or i > b:
                break
            else:
                i += 1

        r1 += " = " + str(int(a)) + "/" + str(int(b))
        print(r1)

File: Ruby/1022 - TDA racional.rb
n = gets.to_i

for i in 1 .. n do 

	info = gets.split(' ')

	n1 = info[0].to_i
	d1 = info[2].to_i
	n2 = info[4].to_i
	d2 = info[6].to_i


	if info[3] == '+' then
		a = n1*d2+n2*d1
		b = d2*d1

		soma = "#{a}/#{b} ="

		j = 2

		while true do

			if a%j == 0 and b%j == 0 then
				a = a/j
				b = b/j
			elsif j > a or j > b then
				break
			else
				j = j + 1
			end
		end

		soma = "#{soma} #{a}/#{b}"

		puts soma
	
	elsif info[3] == '-' then
		a = n1*d2-n2*d1
		b = d2*d1

		soma = "#{a}/#{b} ="

		j = 2

		while true do

			if a%j == 0 and b%j == 0 then
				a = a/j
				b = b/j
			elsif j > a or j > b then
				break
			else
				j = j + 1
			end
		end

		soma = "#{soma} #{a}/#{b}"

		puts soma

	elsif info[3] == '*' then
		a = n1*n2
		b = d1*d2

		soma = "#{a}/#{b} ="

		j = 2

		while true do

			if a%j == 0 and b%j == 0 then
				a = a/j
				b = b/j
			elsif j > a or j > b then
				break
			else
				j = j + 1
			end
		end

		soma = "#{soma} #{a}/#{b}"

		puts soma
	
	elsif info[3] == '/' then
		a = n1*d2
		b = n2*d1

		soma = "#{a}/#{b} ="

		j = 2

		while true do

			if a%j == 0 and b%j == 0 then
				a = a/j
				b = b/j
			elsif j > a or j > b then
				break
			else
				j = j + 1
			end
		end

		soma = "#{soma} #{a}/#{b}"

		puts soma
	end
end